### ISSUE TITLE
Plates with broken thumbnails/images

### ISSUE THREAD

**Issue by sbesson** [IDR/idr0008-rohn-actinome#6]
Captured historically as a Trello card, a few plates have been reported to have broken thumbnails/pixel data in the `screenA` of `idr0008`. In the logs of production IDR, this will manifest itself with errors of type

```
Thumbnail could not be compressed.
java.io.FileNotFoundException: /data/OMERO/Thumbnails/Dir-838/838488 (Read-only file system)
```

Trying to set the rendering setting on this plate gives `ome.conditions.LockTimeout: Import in progress`. Trying to reimport one of the faulty plates (`004B30_S2R`)  results in an `java.lang.RuntimeException: Failure response on import!` with the now informative `Invalid TIFF file: /data/OMERO/ManagedRepository/demo_2/Blitz-0-Ice.ThreadPool.Server-8/2021-04/26/10-49-15.744/004B30_S2R_B22_s1_w1.TIF` message.

The core issue is that some of the TIFF files of the original dataset are corrupted (0B) and cause the reader to fail when trying to read the TIFF header. From a quick audit, 5 plates of `screenA` are affected

```
bash-4.2$ find /uod/idr/filesets/idr0008-rohn-actinome/original-drosophila/Drosophila\ screen/ -type f -size 0
/uod/idr/filesets/idr0008-rohn-actinome/original-drosophila/Drosophila screen/021B30_S2R/021B30_S2R_A14_s1_w1.TIF
/uod/idr/filesets/idr0008-rohn-actinome/original-drosophila/Drosophila screen/021B30_S2R/021B30_S2R_I01_s2_w3.TIF
/uod/idr/filesets/idr0008-rohn-actinome/original-drosophila/Drosophila screen/048B30_S2R/048B30_S2R_H23_s2_w3.TIF
/uod/idr/filesets/idr0008-rohn-actinome/original-drosophila/Drosophila screen/048B30_S2R/048B30_S2R_H23_s1_w3.TIF
/uod/idr/filesets/idr0008-rohn-actinome/original-drosophila/Drosophila screen/026B30_S2R/026B30_S2R_N24_s1_w1.TIF
/uod/idr/filesets/idr0008-rohn-actinome/original-drosophila/Drosophila screen/004B30_S2R/004B30_S2R_B22_s1_w1.TIF
/uod/idr/filesets/idr0008-rohn-actinome/original-drosophila/Drosophila screen/020B30_S2R/020B30_S2R_P23_s2_w1.TIF
```

I think the two routes are:

- try and get back to the original submitter and try to find the true original data
- test removing the 0B files and try a reimport as I think the MetaXpress reader is able to handle missing binary files.

**Comment by sbesson** [comment]
Inspired by a slightly related discussion around the Drosophila gene metadata used in IDR with @dominikl and @francesw , I investigated the second option in a pilot i.e. created a temporary folder symlinking all files of `020B30_S2R`, deleted `020B30_S2R_P23_s2_w1.TIF`.

The plate imported without issue and quite quickly

```
2300 files uploaded, 1 fileset, 1 plate created, 768 images imported, 0 errors in 0:09:22.432
```

The thumbnails render as expected

![Screenshot 2021-09-30 at 17 20 51](https://user-images.githubusercontent.com/1355463/135493460-3cd25089-98d1-4aea-ace4-482782aec320.png)

and the field of view with a missing TIFF has a blank plane as expected

![Screenshot 2021-09-30 at 17 22 57](https://user-images.githubusercontent.com/1355463/135494176-93f181bd-d8ff-411c-8e21-206c21c78a44.png)

Apart from the re-annotation part (which we might want to redo anyways), this feels like a very sane solution to fix one of the few historically broken studies.

**Comment by joshmoore** [comment]
Guess my only thought is if there isn't a way to fix them inline, but full understand that that's probably a lot more than we want to invest. The major benefit I see would be preserving the history information.

**Comment by sbesson** [comment]
Agreed, the major impact of this approach is that any workflow that relies on the image IDs might be broken. Probably worth noting at this stage is that this study will be at least partly re-annotated as per today's decision. 

Fixing the images inline would be ideal and I suspect what needs to happen is first to update the fileset and remove the corrupted TIFF file and then relaunch the import process which is stuck in this `Import in progress`. I have no experience and little time to dedicate to such an investigation unless you have  a few pointers?
